swaylock/shadow.c

```diff
@@ -0,0 +1,124 @@
+#define _XOPEN_SOURCE
```

Comment:
We should probably use `_POSIX_C_SOURCE` instead

Reply:
shadow.h is actually part of the open source extension so here it makes sense